azurerm/internal/services/sentinel/sentinel_automation_rule_resource_test.go

```diff
@@ -0,0 +1,359 @@
+	})
+}
+
+func TestAccSentinelAutomationRule_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
+	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccSentinelAutomationRule_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_sentinel_automation_rule", "test")
+	r := SentinelAutomationRuleResource{uuid: uuid.New().String()}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r SentinelAutomationRuleResource) Exists(ctx context.Context, clients *clients.Client, state *terraform.InstanceState) (*bool, error) {
+	client := clients.Sentinel.AutomationRulesClient
+
+	id, err := parse.AutomationRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp, err := client.Get(ctx, id.ResourceGroup, sentinel.OperationalInsightsResourceProvider, id.WorkspaceName, id.Name); err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(true), nil
+}
+
+func (r SentinelAutomationRuleResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_sentinel_automation_rule" "test" {
+  name                       = "%s"
+  log_analytics_workspace_id = azurerm_log_analytics_solution.sentinel.workspace_resource_id
+  display_name               = "acctest-SentinelAutoRule-%d"
+  order                      = 1
+
+  action_incident {
+    order  = 1
+    status = "Active"
+  }
+}
+`, template, r.uuid, data.RandomInteger)
+}
+
+func (r SentinelAutomationRuleResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+data "azurerm_client_config" "current" {}
+
+data "azuread_service_principal" "securityinsights" {
+  display_name = "Azure Security Insights"
+}
+
+resource "azurerm_role_assignment" "sentinel" {
+  scope                = azurerm_resource_group.test.id
+  role_definition_name = "Azure Sentinel Automation Contributor"
+  principal_id         = data.azuread_service_principal.securityinsights.object_id
+}
+
+resource "azurerm_template_deployment" "testconnection" {
```

Comment:
is there a reason we are not implementing a resource for this? we try our best to never do ARM deployments in acctests

Reply:
This is currently tracked by #1691, which is blocked by lacks of Go SDK support.

Reply:
This is currently tracked by #1691, which is blocked by lacks of Go SDK support.